include/seqan3/alphabet/detail/alphabet_proxy.hpp

```diff
@@ -263,38 +263,41 @@ class alphabet_proxy : public
      *        e.g. seqan3::alphabet_variant.
      * \{
      */
+private:
+    //!\brief work around a gcc bug that disables short-circuiting of operator&& in an enable_if_t of a friend function
+    template <typename t>
+    static constexpr bool is_alphabet_comparable_with = !std::is_same_v<derived_type, t> &&
+                                                        detail::weakly_equality_comparable_with<alphabet_type, t>;
+
```

Comment:
This makes a difference? Interesting.

Reply:
Yes it does. Maybe this fixes your problems where short circuiting didn't work.

Reply:
I moved all the checks out into external variables anyway so I implicitly did the same thing, I think.